test/S3.jl

```diff
@@ -283,7 +283,7 @@ end
                     end
 
                     @testset "Download via presign" begin
-                        src = Path("$s3_prefix/presign/file")
+                        src = S3Path("$s3_prefix/presign/file")
```

Comment:
Might be more spots in this file that need to be updated to `S3Path`

Reply:
That was the only one causing weird test errors, is there a good way to be sure which ones to change if the tests are passing?

Reply:
I think anything that starts with `s3_prefix` can be converted to `S3Path`. Also wondering why this wasn't done in the first place (might be missing something).